lib/pbench/test/unit/server/test_shell_cli.py

```diff
@@ -33,6 +34,88 @@ def exists_true(theself: Path) -> bool:
     return True
 
 
+class FakeIndexer:
+
+    status = 0
+    called = []
+
+    def __init__(self, bin_dir: Path, cwd: Path, logger: logging.Logger):
+        self.bin_dir = bin_dir
+        self.cwd = cwd
+        self.logger = logger
+
+    def start(self):
+        """Start an indexer subprocess.
+
+        The indexer will run in the background, periodically checking for datasets
+        with the INDEX operation enabled. It will loop until it finds no more, and
+        then sleep for a while before trying again.
+
+        Args:
+            bin_dir: Where to find the pbench-index command
+            cwd: Where to set the working directory
+            logger: A Python logger
+
+        Returns:
+            The Popen of the created process so it can be tracked and stopped.
+        """
+        self.called.append("start")
+
+    def shutdown(self):
+        """Stop the indexer
+
+        TODO: We're stopping with SIGTERM, which will interrupt any indexing
+        in progress in favor of a quicker shutdown. We could instead send a
+        """
+
+        self.called.append("shutdown")
+        return self.status
+
+    @classmethod
+    def reset(cls):
+        cls.called.clear()
+        cls.status = 0
+
+
+class FakeEvent:
+    def __init__(self):
+        self.flag = False
+        self.waited = 0
+
+    def set(self):
+        self.flag = True
+
+    def wait(self, timeout: float) -> bool:
+        self.waited += 1
+        return self.flag
+
+
+@pytest.fixture
+def mock_indexer(monkeypatch, server_config):
+    FakeIndexer.reset()
+    monkeypatch.setattr("pbench.cli.server.shell.PbenchIndexer", FakeIndexer)
+
+
+class FakePopen:
+
+    processes = {}
+    pid = 0
+
+    def __init__(self, args: list[Union[str, Path]], cwd: Path):
+        self.cmd = " ".join(str(a) for a in args)
+        self.running = True
+        self.waited = False
+        __class__.pid += 1
+        self.pid = __class__.pid
+        self.processes[self.pid] = self
+
+    def terminate(self):
+        self.running = False
+
+    def wait(self, timeout: float = 0.0):
+        self.waited = True
+
```

Comment:
[`Popen.wait()`](https://docs.python.org/3/library/subprocess.html#subprocess.Popen.wait) is supposed to return a value (if it doesn't time out).